slatedb/src/compactor.rs

```diff
@@ -476,64 +467,46 @@ impl CompactorEventHandler {
     }
 
     async fn maybe_schedule_compactions(&mut self) -> Result<(), SlateDBError> {
-        let compactor_job_requests = self.scheduler.maybe_schedule_compaction(&self.state);
-        for compactor_job_request in compactor_job_requests.iter() {
-            if self.state.num_compactions() >= self.options.max_concurrent_compactions {
+        let mut specs = self.scheduler.maybe_schedule_compaction(&self.state);
+        for spec in specs.drain(..) {
+            let active_compactions = self.state.jobs().count();
+            if active_compactions >= self.options.max_concurrent_compactions {
                 info!(
                     "already running {} compactions, which is at the max {}. Won't run compaction {:?}",
-                    self.state.num_compactions(),
+                    active_compactions,
                     self.options.max_concurrent_compactions,
-                    compactor_job_request
+                    spec
                 );
                 break;
             }
-            // Pass compactor job in here
-            let compactor_job_id = self.rand.rng().gen_ulid(self.system_clock.as_ref());
-            let job_input = CompactorJobInput::SortedRunJobInputs {
-                ssts: CompactorJob::get_ssts(
-                    self.state.db_state(),
-                    compactor_job_request.sources(),
-                ),
-                sorted_runs: CompactorJob::get_sorted_runs(
-                    self.state.db_state(),
-                    compactor_job_request.sources(),
-                ),
-            };
-            let compactor_job: CompactorJob = CompactorJob::new(
-                compactor_job_id,
-                CompactorJobRequestType::Internal,
-                compactor_job_request.clone(),
-                job_input,
-            );
-            self.submit_compaction(compactor_job).await?;
+            let job_id = self.rand.rng().gen_ulid(self.system_clock.as_ref());
+            let job = CompactorJob::new(job_id, spec);
+            self.submit_compaction(job).await?;
         }
         Ok(())
     }
 
     async fn start_compaction(
         &mut self,
-        id: Ulid,
-        compactor_job: CompactorJob,
+        attempt_id: Ulid,
+        job: CompactorJob,
     ) -> Result<(), SlateDBError> {
         self.log_compaction_state();
         let db_state = self.state.db_state();
-        let request = compactor_job.compactor_job_request();
-        let (ssts, sorted_runs) = match &compactor_job.job_input() {
-            CompactorJobInput::SortedRunJobInputs { ssts, sorted_runs } => {
-                (ssts.to_vec(), sorted_runs.to_vec())
-            }
-        };
+        let spec = job.spec();
```

Comment:
this is a bit weird - why not just make `get_ssts` definition take `&self` rather than pulling out the spec and then passing it back in

Reply:
Agh, that was just brainless copy/pasting. My bad. 1a975abc